### PR TITLE
devcontainer: Update and add support for arm64 hosts

### PR DESCRIPTION
**Pull request by barrbrain**
This will facilitate rebuild when a new Rust toolchain is released. We should sync with `cargo-c` releases.
Added support for `linux/arm64` hosts to the prebuilt image.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/devcontainers/rust:1-1-bookworm
+FROM mcr.microsoft.com/devcontainers/rust:1.0.5-1-bookworm
```
Comment (barrbrain) [inline .devcontainer/Dockerfile:1]: Alternatively, pin the the MSRV 1.70.0. See https://github.com/devcontainers/images/blob/main/src/rust/history/1.0.1.md#variant-bookworm
```suggestion
FROM mcr.microsoft.com/devcontainers/rust:1.0.1-1-bookworm
```

**Comment by barrbrain** [comment]
I might need to rework this to avoid pulling in deps with open CVEs.

**Comment by barrbrain** [comment]
I verified the muilt-arch manifest on my fork:
```sh
docker manifest inspect ghcr.io/barrbrain/rav1e-devcontainer:latest
```
```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 3906,
         "digest": "sha256:b6a67286b6299678c40e9e6abd029dbac8da2382cb10c7e04c984532ae1dfa0b",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 3694,
         "digest": "sha256:f6e81cc7caadfff3e121c70daf70c48cddb72efa977873600d29237385dd6bfe",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

**Comment by barrbrain** [comment]
We could save a lot of build time by cross-compiling the cargo tools:
```
apt-get install libcurl4-openssl-dev:arm64 libz-dev:arm64
export PKG_CONFIG_PATH=/usr/lib/aarch64-linux-gnu/pkgconfig
export CARGO_PROFILE_RELEASE_STRIP=true
export CARGO_BUILD_TARGET=aarch64-unknown-linux-gnu
cargo install -q cargo-c --version "0.9.27+cargo-0.74.0"
cargo install -q cargo-criterion --version "1.1.0"
cargo install -q cargo-fuzz --version "0.11.2"
```

**Review by lu-zero — APPROVED** [review]

**Pull request merged**